### DISCUSSION TITLE
Question: Is it possible to build Drive service with service account without key file?

### DISCUSSION THREAD

**Discussion by dzahedia** [googleapis/google-api-python-client#2024] [Q&A]
I use cloud functions and can connect to bigquery or cloud storage without key files. Is it possible to do the same with Drive service? In our company, we are not allowed to create service account key files. Thanks!

**Comment by boorge** [comment]
Yes, it is possible to interact with Google Drive using a service account without explicitly using a key file. Google Cloud offers a feature called **Application Default Credentials** (ADC) that can simplify authentication in certain environments, such as Cloud Functions.

**Reply by safsz** [reply]
Can you elaborate more on this? Especially how to implement this when using cloud functions
